src/utils/net.rs

```diff
@@ -18,6 +22,34 @@ pub fn global_http_client() -> reqwest::Client {
     CLIENT.clone()
 }
 
+/// Download a file via IPFS HTTP gateway in trustless mode.
+/// See <https://github.com/ipfs/specs/blob/main/http-gateways/TRUSTLESS_GATEWAY.md>
+pub async fn download_ipfs_file_trustlessly(
+    cid: &Cid,
+    gateway: Option<&str>,
+    destination: &Path,
+) -> anyhow::Result<()> {
+    // https://docs.ipfs.tech/concepts/ipfs-gateway/
+    const DEFAULT_IPFS_GATEWAY: &str = "https://ipfs.io/ipfs/";
+
+    let url = format!(
+        "{}{cid}?format=car",
+        gateway.unwrap_or(DEFAULT_IPFS_GATEWAY)
+    );
```

Comment:
I wonder if we can use the methods of an `Url` struct directly without string manipulation?

Reply:
Url crate does not really provide a builder API, I've switched to `join` and `set_query` to make it better than string manipulation